# Data Science Project: Profitable App Profiles#

In this project, I will be pretending to be a data analyst that works for an app developer company.

The company only makes free apps and generates its revenue via ads. The goal of the company is to have as many users of the apps because revenue is based off of how many people see an ad. 

The goal of the project is to demonstrate what types of apps are more likely to be successful in attaining a large amount of users.


Let me know if you have any questions! You may email me at: robertchasesommer@gmail.com


I will be using the Google Play Store and Apple datasets from Kaggle.

Google: https://www.kaggle.com/lava18/google-play-store-apps/home

Apple: https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home

# Step 1: Loading the data

This is code I use to open my files:

In [1]:
from csv import reader

opened_file1 = open('AppleStore.csv', encoding='utf8')
read_file1 = reader(opened_file1)
Apple = list(read_file1)

opened_file2 = open('googleplaystore.csv', encoding='utf8')
read_file2 = reader(opened_file2)
Google = list(read_file2)

The function below was created by DataQuest. It is used to explore a dataset. 

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [3]:
explore_data(Apple, 0, 3)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']




In [4]:
explore_data(Google, 0 ,3)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']




By looking at the headers, I am able to see that there are similar columns but with different names. Here is a chart of similar names for the same values.

|Apple | Google|
|---|---|
|track_name | App|
|size_bytes | Size|
|price | Price|
|user_rating | Rating|
|cont_rating | Content Rating|
|prime_genre | Genres|

This list is not meant to be exhaustive, but it meant to show that there are similar columns that we may be able to join the two different datasets on. How cool!




# Step 2: Cleaning the data

Now remember, the company that I work for is looking to create free apps in the english language. So, I will need to isolate the free and english apps in a new data set. 

Also, I will want to remove any duplicates. 

Looking at the discussion page on Kaggle, I can also see that there is a row that is wrong (entry 10472) on the Google dataset. I will want to fix that! 


In [5]:
print(Google[10473]) 

"The value after 'Everyone' is blank. I will delete this column."
print(len(Google))
del(Google[10473])
print(len(Google))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10842
10841


The code below demonstrates that there are duplicates in the dataset. Specifically, Instagram is shown four times.

In [6]:
for app in Google:
    appname = app[0]
    if appname == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


The below code demonstrates that there are many duplicate rows. In fact, there are 1181 duplicates!

In [7]:
distinct_apps = []
repeated_apps = []

for app in Google:
    app_name = app[0]
    if app_name in distinct_apps:
        repeated_apps.append(app_name)
    else:
        distinct_apps.append(app_name)

print('Number of distinct apps', len(distinct_apps))
print('Number of repeated apps', len(repeated_apps))

Number of distinct apps 9660
Number of repeated apps 1181


Going back to the code that demonstrates that there are duplicates of the Instagram app, let's see if there is a difference between the duplicate rows.

In [8]:
for app in Google:
    appname = app[0]
    if appname == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


As we can see, the number of ratings is different for each duplicate row. This might indicate that the person who scraped this data may have attempted more than once, or maybe the web scrapper malfunctioned. Either way, we need to remove the duplicates.

I will not randomly pick which duplicates to delete. Instead, I will keep the duplicates with the highest number of ratings. The highest number of ratings implies that it was scrapped most recently.

Going back the the Instagram example, I would keep the second row that has `66577446` ratings because it has the most ratings. 

In [9]:
reviews_max = {}

for row in Google[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
print(len(reviews_max))
    
    

9659


The `reviews_max` dictionary will be used as a tool to create a clean dataset with no repeating rows. It is not the final dataset.

The `reviews_max` will be used like a measuring stick to help find all of the rows witht the highest rating counts.

Below is a clean-er Google dataset! As a recap, we removed all duplicate rows in the Google dataset. 

In [10]:
Google_clean = []
Google_added = []

for row in Google[1:]:
    name = row[0]
    n_reviews = float(row[3])
    if (n_reviews == reviews_max[name]) and (name not in Google_added):
        Google_clean.append(row)
        Google_added.append(name)
        
explore_data(Google_clean, 0,3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


Remember, we are a company that is looking to develop English apps. So guess what we are doing next? Correct! Removing all non-english apps from the dataset. 

### Teaching session! ASCII

So, in order to find all non-english apps we need to search for apps that contain non-english characters in the titles. How do we do this? First we need to have a brief understanding of ASCII (American Standard Code for Information Interchange).

Each character in the alphabet has a number associated with it:
a = 97
A = 65
5 = 53

All English characters fall in the range from 0 to 127. A non english character would have a number greater than 127. For example:

爱 = 29233

Pretty interesting, right??

Continuing with our project, we will want to find the apps with titles that contain characters that fall in the range of 0 to 127.

We will use the function `ord()` to do this.

In [11]:
def is_english(string):
    
    for character in string:
        if ord(character) > 127:
            return False
        
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

True
False
False
False
8482
128540


Notice that the 3rd and 4th entries are labelled as non-english. This is because emojies are not in the ASCII. To correct this we will change the function to only `return false` when there are 3 or more 127+ symbols. 

In [12]:
def is_english(string):
    over = 0
    for character in string:
        if ord(character) > 127:
            over += 1
        if over >3:
            return False
        
    return True

print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

True
False
True
True
8482
128540


It worked! Now let's use the `is_english()` function on both datasets to filter out non-english apps.

In [13]:
Google_clean_english = []
Apple_english = []

for row in Google_clean:
    name = row[0]
    if is_english(name):
        Google_clean_english.append(row)
        
for row in Apple[1:]:
    name = row[2]
    if is_english(name):
        Apple_english.append(row)
        
explore_data(Google_clean_english, 0, 3, True)
print('\n')
explore_data(Apple_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188

After filtering out non-english apps, we can see that there are 9614 apps in the Google Play Store and 6183 apps in the Apple Store. Cool beans.

#### Recap
Ok, so far we have:
    1. Removed inaccurate/distorted rows
    2. Removed duplicate apps
    3. Removed non-english apps

Now, we will want to remove the non-free apps. Remember that our company only makes english, free apps.

In [14]:
print(Google[0])
print('\n')
print(Apple[0])

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


In [15]:
Google_free = []
Apple_free = []

for row in Google_clean_english:
    price = row[7]
    if price == '0':
        Google_free.append(row)
print(len(Google_free))
    
    
    
for row in Apple_english:
    price = float(row[5])
    if price == 0:
        Apple_free.append(row)
print(len(Apple_free))

8864
3222


# Part 3: Analyzing Data

Remember, our goal is to create an app that is most likely to have a large amount of users.

Narrowing down our approach, we will:
1. Build a minimal version of an app on the Google Play Store.
2. If the app has a good response, we will develop it further.
3. If the app is profitable after 6 months, we will create an iOS version.

We will need to see what types of apps are likely to be a success on both Google and Apple stores. 

We will begin by looking at the most common genres of apps using frequency tables. 

The code below converts a dataset and the specified column into a frequency table. Then the frequency table is converted to percentages because it is easier to read this way. 

In [16]:
def freq_table(dataset, index):
    table = {}
    total = 0
    
    for row in dataset:
        total += 1
        column = row[index]
        if column in table:
            table[column] += 1
        else:
            table[column] = 1
    
    table_percentages = {}
    for key in table:
        percentage = round(((table[key] / total) * 100),2)
        table_percentages[key] = percentage
        
    return table_percentages
        
            
    
    

In [17]:
freq_table(Google_free, 9)

{'Art & Design': 0.6,
 'Art & Design;Creativity': 0.07,
 'Auto & Vehicles': 0.93,
 'Beauty': 0.6,
 'Books & Reference': 2.14,
 'Business': 4.59,
 'Comics': 0.61,
 'Comics;Creativity': 0.01,
 'Communication': 3.24,
 'Dating': 1.86,
 'Education': 5.35,
 'Education;Creativity': 0.05,
 'Education;Education': 0.34,
 'Education;Pretend Play': 0.06,
 'Education;Brain Games': 0.03,
 'Entertainment': 6.07,
 'Entertainment;Brain Games': 0.08,
 'Entertainment;Creativity': 0.03,
 'Entertainment;Music & Video': 0.17,
 'Events': 0.71,
 'Finance': 3.7,
 'Food & Drink': 1.24,
 'Health & Fitness': 3.08,
 'House & Home': 0.82,
 'Libraries & Demo': 0.94,
 'Lifestyle': 3.89,
 'Lifestyle;Pretend Play': 0.01,
 'Card': 0.45,
 'Arcade': 1.85,
 'Puzzle': 1.13,
 'Racing': 0.99,
 'Sports': 3.46,
 'Casual': 1.76,
 'Simulation': 2.04,
 'Adventure': 0.68,
 'Trivia': 0.42,
 'Action': 3.1,
 'Word': 0.26,
 'Role Playing': 0.94,
 'Strategy': 0.91,
 'Board': 0.38,
 'Music': 0.2,
 'Action;Action & Adventure': 0.1,
 'Casu

The code below is copied from Dataquest. It uses the function `freq_table` and sorts the frequencies by descending order.

In [18]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [19]:
display_table(Google_free, 9) #Genres

Tools : 8.45
Entertainment : 6.07
Education : 5.35
Business : 4.59
Productivity : 3.89
Lifestyle : 3.89
Finance : 3.7
Medical : 3.53
Sports : 3.46
Personalization : 3.32
Communication : 3.24
Action : 3.1
Health & Fitness : 3.08
Photography : 2.94
News & Magazines : 2.8
Social : 2.66
Travel & Local : 2.32
Shopping : 2.25
Books & Reference : 2.14
Simulation : 2.04
Dating : 1.86
Arcade : 1.85
Video Players & Editors : 1.77
Casual : 1.76
Maps & Navigation : 1.4
Food & Drink : 1.24
Puzzle : 1.13
Racing : 0.99
Role Playing : 0.94
Libraries & Demo : 0.94
Auto & Vehicles : 0.93
Strategy : 0.91
House & Home : 0.82
Weather : 0.8
Events : 0.71
Adventure : 0.68
Comics : 0.61
Beauty : 0.6
Art & Design : 0.6
Parenting : 0.5
Card : 0.45
Casino : 0.43
Trivia : 0.42
Educational;Education : 0.39
Board : 0.38
Educational : 0.37
Education;Education : 0.34
Word : 0.26
Casual;Pretend Play : 0.24
Music : 0.2
Racing;Action & Adventure : 0.17
Puzzle;Brain Games : 0.17
Entertainment;Music & Video : 0.17
Casual;

By looking at this frequency table, I can see that 
1. Tools is the biggest category. 
2. The difference between categories is not anything surprising. 
3. The categories could be skewed because there are so many sub-categories.

Based on this frequency table alone, I would recommend a Tool, Entertainment, Education, or Business app. However, I will need to see whether #1 on the frequency table is related to high numbers of users. It may not be the case.

In [20]:
display_table(Google_free, 1) #Category

FAMILY : 18.91
GAME : 9.72
TOOLS : 8.46
BUSINESS : 4.59
LIFESTYLE : 3.9
PRODUCTIVITY : 3.89
FINANCE : 3.7
MEDICAL : 3.53
SPORTS : 3.4
PERSONALIZATION : 3.32
COMMUNICATION : 3.24
HEALTH_AND_FITNESS : 3.08
PHOTOGRAPHY : 2.94
NEWS_AND_MAGAZINES : 2.8
SOCIAL : 2.66
TRAVEL_AND_LOCAL : 2.34
SHOPPING : 2.25
BOOKS_AND_REFERENCE : 2.14
DATING : 1.86
VIDEO_PLAYERS : 1.79
MAPS_AND_NAVIGATION : 1.4
FOOD_AND_DRINK : 1.24
EDUCATION : 1.16
ENTERTAINMENT : 0.96
LIBRARIES_AND_DEMO : 0.94
AUTO_AND_VEHICLES : 0.93
HOUSE_AND_HOME : 0.82
WEATHER : 0.8
EVENTS : 0.71
PARENTING : 0.65
ART_AND_DESIGN : 0.64
COMICS : 0.62
BEAUTY : 0.6


This table seems easier to read. I am a bit confused as to why Genres and Category are so different. I will need to investigate more as to why there is a need for Category and Genre.

In [21]:
display_table(Apple_free, 12) #prime_genre

Games : 58.16
Entertainment : 7.88
Photo & Video : 4.97
Education : 3.66
Social Networking : 3.29
Shopping : 2.61
Utilities : 2.51
Sports : 2.14
Music : 2.05
Health & Fitness : 2.02
Productivity : 1.74
Lifestyle : 1.58
News : 1.33
Travel : 1.24
Finance : 1.12
Weather : 0.87
Food & Drink : 0.81
Reference : 0.56
Business : 0.53
Book : 0.43
Navigation : 0.19
Medical : 0.19
Catalogs : 0.12


Gaming seems to be high on the list for both Google and Apple. Apple, however has significantly more gaming apps than Google. Interesting, I wonder why. 

### Most Popular Genres: Genres with the most users

Now we are going to find out which genres are the most popular. To do this, we will measure the number of `Installs` on the Google dataset and will use `rating_count_tot` for the Apple dataset. Apple does not have an Installs column, so we are using a workaround. We are assuming that if someone has rated an app, they are also a user. Therefore, `rating_count_tot` will be a good indicator of popularity. 

We will start by calculating the average number of user ratings per app-genre on the Apple App store. To do this we will need to:
1. Isolate the apps of each genre
2. Sum up the user ratings for each app of that genre.
3. Divide the sum by the number of apps belonging to that genre

In [26]:
freq_table(Apple_free, 12)

{'Productivity': 1.74,
 'Weather': 0.87,
 'Shopping': 2.61,
 'Reference': 0.56,
 'Finance': 1.12,
 'Music': 2.05,
 'Utilities': 2.51,
 'Travel': 1.24,
 'Social Networking': 3.29,
 'Sports': 2.14,
 'Health & Fitness': 2.02,
 'Games': 58.16,
 'Food & Drink': 0.81,
 'News': 1.33,
 'Book': 0.43,
 'Photo & Video': 4.97,
 'Entertainment': 7.88,
 'Business': 0.53,
 'Lifestyle': 1.58,
 'Education': 3.66,
 'Navigation': 0.19,
 'Medical': 0.19,
 'Catalogs': 0.12}

Below is the average number of user ratings per genre on the Apple Apps Store.

In [29]:
Apple_genres = freq_table(Apple_free, 12)

for genre in Apple_genres:
    total = 0
    len_genre = 0 #This will store the number of apps in each genre.
    
    for iteration in Apple_free:
        genre_app = iteration[12]
        if genre_app == genre:
            ratings = float(iteration[6])
            total += ratings
            len_genre += 1
    average = round((total / len_genre),2)
    print(genre, ":", average)
   
            
        
    

Productivity : 21028.41
Weather : 52279.89
Shopping : 26919.69
Reference : 74942.11
Finance : 31467.94
Music : 57326.53
Utilities : 18684.46
Travel : 28243.8
Social Networking : 71548.35
Sports : 23008.9
Health & Fitness : 23298.02
Games : 22788.67
Food & Drink : 33333.92
News : 21248.02
Book : 39758.5
Photo & Video : 28441.54
Entertainment : 14029.83
Business : 7491.12
Lifestyle : 16485.76
Education : 7003.98
Navigation : 86090.33
Medical : 612.0
Catalogs : 4004.0


Navigation seems to be the genre with the largest amount of ratings. However, when looking into the the specific genres, it seems that two apps, Google Maps and Waze, has significantly more than the rest. This means that the Navigation apps are either boom or bust and is not a reliable genre to pursue.

In [33]:
for iteration in Apple_free:
    if iteration[12] == 'Navigation':
        print(iteration[12],iteration[2],":",iteration[6])

Navigation Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Navigation Geocaching® : 12811
Navigation ImmobilienScout24: Real Estate Search in Germany : 187
Navigation Railway Route Search : 5
Navigation CoPilot GPS – Car Navigation & Offline Maps : 3582
Navigation Google Maps - Navigation & Transit : 154911


In [43]:
Google_genres = freq_table(Google_free, 1)

for category in Google_genres:
    total = 0
    len_category = 0
    
    for iteration in Google_free:
        category_app = iteration[1]
        if category_app == category:
            installs = iteration[5]
            installs = installs.replace('+','')
            installs = installs.replace(',','')
            installs = float(installs)
            total += installs
            len_category += 1
    average = round((total/len_category),2)
    print(category,":",average)
            
            
            

ART_AND_DESIGN : 1986335.09
AUTO_AND_VEHICLES : 647317.82
BEAUTY : 513151.89
BOOKS_AND_REFERENCE : 8767811.89
BUSINESS : 1712290.15
COMICS : 817657.27
COMMUNICATION : 38456119.17
DATING : 854028.83
EDUCATION : 1833495.15
ENTERTAINMENT : 11640705.88
EVENTS : 253542.22
FINANCE : 1387692.48
FOOD_AND_DRINK : 1924897.74
HEALTH_AND_FITNESS : 4188821.99
HOUSE_AND_HOME : 1331540.56
LIBRARIES_AND_DEMO : 638503.73
LIFESTYLE : 1437816.27
GAME : 15588015.6
FAMILY : 3695641.82
MEDICAL : 120550.62
SOCIAL : 23253652.13
SHOPPING : 7036877.31
PHOTOGRAPHY : 17840110.4
SPORTS : 3638640.14
TRAVEL_AND_LOCAL : 13984077.71
TOOLS : 10801391.3
PERSONALIZATION : 5201482.61
PRODUCTIVITY : 16787331.34
PARENTING : 542603.62
WEATHER : 5074486.2
VIDEO_PLAYERS : 24727872.45
NEWS_AND_MAGAZINES : 9549178.47
MAPS_AND_NAVIGATION : 4056941.77
